### DISCUSSION TITLE
the trait `From<Vec<std::string::String>>` is not implemented for `sea_orm::Value`

### DISCUSSION THREAD

**Discussion by road2react** [SeaQL/sea-orm#1596] [Q&A]
## Description

the trait `From<Vec<std::string::String>>` is not implemented for `sea_orm::Value`

## Steps to Reproduce

1. Create a PostgreSQL database with a table
2. Add a nullable column to the table that contains an array of strings
3. Run `sea-orm-cli generate entity`

### Expected Behavior

The generated entity files should work.

### Actual Behavior

The generated struct looks like this (with `--expanded-format`):

```
#[derive(Clone, Debug, PartialEq, DeriveModel, DeriveActiveModel, Serialize, Deserialize)]
pub struct Model {
    pub column_with_array: Option<Vec<String>>,
}
```

However, this leads to a compile error:

```
    | ...Eq, DeriveModel, DeriveActiveModel, Serialize, Deserialize)]
    |                     ^^^^^^^^^^^^^^^^^
    |                     |
    |                     the trait `From<Vec<std::string::String>>` is not implemented for `sea_orm::Value`
    |                     in this derive macro expansion
```

### Reproduces How Often

Always happens

### Workarounds

The generated type could be changed, but it would no longer exactly match the database.

## Versions

`sea-orm-cli 0.11.1`

`sea-orm 0.11.1`

**Comment by billy1624** [comment]
Hey @road2react, welcome to SeaQL. A sanity check: you need to enable `postgres-array` feature on `sea-orm` for it to work.

**Comment by Nueck** [comment]
@billy1624 I try to use  `postgres-array`  but I don't know how to retrieve `text[]`

**Reply by billy1624** [reply]
Sorry, I don't quite understand the problem. Could you please elaborate?

**Reply by Nueck** [reply]
I want to use the text[] in postgres to store the array. Want to use the filter in sea-orm to achieve conditional judgment. 
But I don't know how to do that. Here is a query
`SELECT * FROM "article" WHERE "article"."user_id" = 1  AND 'data'=ANY(tag) ORDER BY "article"."article_id" DESC`
Use the filter in sea-orm for syntax-like functionality

**Reply by billy1624** [reply]
```rs
Entity::find()
    .filter(Into::<SimpleExpr>::into(PgFunc::any( ... )))
```

<https://docs.rs/sea-query/latest/sea_query/extension/postgres/struct.PgFunc.html#method.any>
